src/dbnode/storage/bootstrap/types.go

```diff
@@ -50,13 +52,97 @@ type ProcessProvider interface {
 // with the mindset that it will always be set to default values from the constructor.
 type Process interface {
 	// Run runs the bootstrap process, returning the bootstrap result and any error encountered.
-	Run(start time.Time, ns namespace.Metadata, shards []uint32) (ProcessResult, error)
+	Run(start time.Time, namespaces []ProcessNamespace) (NamespaceResults, error)
+}
+
+// ProcessNamespace is a namespace to pass to the bootstrap process.
+type ProcessNamespace struct {
+	// Metadata of the namespace being bootstrapped.
+	Metadata namespace.Metadata
+	// Shards is the shards to bootstrap for the bootstrap process.
+	Shards []uint32
+	// DataAccumulator is the data accumulator for the shards.
+	DataAccumulator NamespaceDataAccumulator
+}
+
+// Namespaces are a set of namespaces being bootstrapped.
+type Namespaces struct {
+	Namespaces *NamespacesMap
+}
+
+// Namespace is a namespace that is being bootstrapped.
+type Namespace struct {
+	// Metadata of the namespace being bootstrapped.
+	Metadata namespace.Metadata
+	// Shards is the shards for the namespace being bootstrapped.
+	Shards []uint32
+	// DataAccumulator is the data accumulator for the shards.
+	DataAccumulator NamespaceDataAccumulator
+	// DataTargetRange is the data target bootstrap range.
+	DataTargetRange TargetRange
+	// IndexTargetRange is the data target bootstrap range.
+	IndexTargetRange TargetRange
+	// DataRunOptions is a set of run options for the current
+	// series data bootstrap run.
+	DataRunOptions NamespaceRunOptions
+	// IndexRunOptions is a set of run options for the current
+	// series index metadata bootstrap run.
+	IndexRunOptions NamespaceRunOptions
+}
+
+// NamespaceRunOptions is the namespace run options for the current
```

Comment:
nit:`// NamespaceRunOptions are the run options for a bootstrap process run.`

Reply:
Sounds good.